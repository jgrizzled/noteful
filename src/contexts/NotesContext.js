import React from 'react';

export default React.createContext({
  notes: [],
  folders: [],
  addNote: () => null,
  deleteNote: () => null,
  addFolder: () => null
});
