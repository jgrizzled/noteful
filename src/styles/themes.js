// light and dark themes

const baseTheme = {
  fontSize: {
    small: '0.85rem',
    normal: '1rem',
    large: '1.25rem',
    xLarge: '1.5rem',
    xxLarge: '2rem'
  },
  font: {
    main: "'Roboto', sans-serif;"
  },
  fontWeight: {
    normal: '400',
    bold: '700'
  },
  breakpoint: {
    medium: '600px',
    large: '1000px'
  }
};

const lightTheme = {
  ...baseTheme,
  color: {
    primary: '#2bbd7e',
    background: '#FFFFFF',
    surface: '#f2f2f2',
    error: '#B00020',
    onPrimary: '#000000',
    onBackground: '#000000',
    onSurface: '#000000',
    link: '#1890ff'
  },
  boxShadow: {
    normal: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
    normalHover: '0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)'
  }
};

const darkTheme = {
  ...baseTheme,
  color: {
    primary: '#2bbd7e',
    background: '#121212',
    surface: '#333333',
    error: '#CF6679',
    onPrimary: '#000000',
    onBackground: '#FFFFFF',
    onSurface: '#FFFFFF',
    link: '#99ceff'
  },
  boxShadow: {
    normal:
      '0 1px 3px rgba(255, 255, 255, 0.25), 0 1px 2px rgba(255, 255, 255, 0.25)',
    normalHover:
      '0px 5px 14px rgba(255, 255, 255, 0.5), 0px 3px 5px rgba(255, 255, 255, 0.22)'
  }
};

export default {
  light: lightTheme,
  dark: darkTheme
};
