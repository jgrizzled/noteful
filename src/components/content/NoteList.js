// list of NoteCards in folder

import React, { useContext } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import styled from 'styled-components';
import NoteCard from 'components/content/NoteCard';
import NotesContext from 'contexts/NotesContext';
import Button from 'components/common/Button';
import ErrorMessage from 'components/common/ErrorMessage';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const NoteList = () => {
  const params = useParams();
  const history = useHistory();
  const { notes, folders } = useContext(NotesContext);
  let renderNotes = notes;
  let folderId = '';
  if (params.folderId) {
    const matchFolder = folders.find(f => String(f.id) === params.folderId);
    if (!matchFolder) return <ErrorMessage>Folder not found!</ErrorMessage>;
    renderNotes = notes.filter(n => String(n.folderId) === params.folderId);
    folderId = params.folderId;
  }
  return (
    <Container>
      <Button large onClick={() => history.push(`/addnote/${folderId}`)}>
        Add Note
      </Button>
      {renderNotes.map((n, i) => (
        <NoteCard note={n} key={i} />
      ))}
    </Container>
  );
};

export default NoteList;
