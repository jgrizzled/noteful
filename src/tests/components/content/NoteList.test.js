import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import themes from 'styles/themes';
import NoteList from 'components/content/NoteList';
import NotesContext from 'contexts/NotesContext';

const component = (
  <BrowserRouter>
    <NotesContext.Provider value={notesContextValue}>
      <ThemeProvider theme={themes.light}>
        <NoteList />
      </ThemeProvider>
    </NotesContext.Provider>
  </BrowserRouter>
);

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(component, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
