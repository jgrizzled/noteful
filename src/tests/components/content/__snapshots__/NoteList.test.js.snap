// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="sc-AykKG iXeyVM"
>
  <button
    className="sc-AykKC idWUwc"
    onClick={[Function]}
  >
    Add Note
  </button>
  <section
    className="sc-AykKE dMUvDE"
    onClick={[Function]}
  >
    <h2>
      test note
    </h2>
    <p>
      Last modified on 
      Invalid date
      <button
        className="sc-AykKC sc-AykKD gVWOnP"
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </section>
</div>
`;
